knowledge_repo/app/routes/posts.py

```diff
@@ -172,3 +177,33 @@ def render_legacy():
 def about():
     """Renders about page. This is the html version of REAMDE.md"""
     return render_template("about.html")
+
+
+@blueprint.route('/ajax/post/download', methods=['GET'])
+@PageView.logged
+def download():
+    "Downloads resources associated with a post."
+
+    path = request.args.get('post')
+    post = current_repo.post(path)
+
+    resource_type = request.args.get('type', 'source')
+    post = current_repo.post(path)
+
+    if resource_type in ('kp', 'zip'):
```

Comment:
quesiton: so is a .kp format the zip of the directory structure of a knowledge post?